packages/moonstone/Popup/Popup.js

```diff
@@ -453,9 +452,11 @@ class Popup extends React.Component {
 
 	handleFloatingLayerOpen = () => {
 		if (!this.props.noAnimation) {
-			this.setState({
-				popupOpen: OpenState.OPENING
-			});
+			if (this.state.popupOpen !== OpenState.OPEN) {
```

Comment:
Can you combine this condition and the one above? Would there be issues with doing so?

```
if (!this.props.noAnimation && this.state.popupOpen !== OpenState.OPEN) {
```

Reply:
No. The condition above looks better.

Reply:
I agree with @ybsung since `spotPopupContent` will be called within `componentDidMount` then also will be called in below _else if_ clause when (!this.props.noAnimation && this.state.popupOpen === OpenState.OPEN && this.props.open) is true.

Reply:
Won't that still be the case with the code as is? I'll need to step through it tomorrow to verify.

Reply:
I tested code in this PR briefly with qa-sampler, and `spotPopupContent` is called twice due to calling in `handlePopupShow` (not `handleFloatingLayerOpen`) which is called back from `PopupBase`. Hmm...